### ISSUE TITLE
Template dir for local development

### ISSUE THREAD

**Issue by bodgerbarnett** [ecometrica/django-multisite#53]
I am appending .local to my domain so that I can develop locally but that means I must change the domain of each site in the database to have .local on the end of it.

This, in turn, means that the template dir that is used is wrong. It is, for example, example.com.local, but the templates are in example.com.

Is there any way around this?

**Comment by rebkwok** [comment]
Do you mean that you have a custom template subfolders in your top-level templates dir that you've named 'example.com' and  I guess you are looking for templates to render in your views by `{site.domain}/foo.html`?  
Can you keep the domain for your site as it is (example.com) and create an alias with 
 the domain you want to use for local development (example.com.local)?  Then your template lookup will remain the same, because your site domain is still example.com.

**Comment by bodgerbarnett** [comment]
I'm not sure what you mean by "create an alias with the domain you want to use for local development" in this case?

**Comment by rebkwok** [comment]
Create an alias in the django admin. Or:
```
site = Site.objects.create(name='example site', domain='example.com')
alias = Alias.objects.create(site=site, domain='example.com.local')
```
Use url example.com.local for your local development; since it is an alias for example.com, your template lookups that are based on `site.domain` will still point to the same location.

I'm not sure why you need to append .local to your site domains for local development, but this will allow you to use a different url locally but keep the site domain the same.  In my local development with multisite, I just add an entry in my `/etc/hosts` file for the site domain, as described in the README.

**Comment by bodgerbarnett** [comment]
OK, I will try the alias thing, thanks.

The reason I use .local for local development (rather than using /etc/hosts) is that if I use /etc/hosts, that removes access to the real domain so I can't check the actual live site at any point, without editing the hosts file. Maybe there's a way round that but I'm not aware of one.

**Comment by rebkwok** [comment]
Assuming you are running a local django server and accessing it at http://example.com:8000 (or whatever port you are running on), adding it to /etc/hosts doesn't prevent you using the real domain (without specified port).

**Comment by rebkwok** [comment]
Closing since this is not an issue with multisite itself

**Comment by bodgerbarnett** [comment]
Yup, fair enough. The odd thing though, is that if I put example.com in the /etc/hosts file, it resolves to 127.0.0.1 for the real domain as well as the example.com:8000 (local dev) one. I guess I must be doing something wrong.
